mod/ping.php

```diff
@@ -401,12 +401,15 @@ function ping_get_notifications($uid)
 	do {
 		$r = DBA::toArray(DBA::p(
 			"SELECT `notify`.*, `post`.`visible`, `post`.`deleted`
-			FROM `notify` LEFT JOIN `post` ON `post`.`uri-id` = `notify`.`uri-id`
-			WHERE `notify`.`uid` = ? AND `notify`.`msg` != ''
-			AND NOT (`notify`.`type` IN (?, ?))
-			AND $seensql `notify`.`seen` ORDER BY `notify`.`date` $order LIMIT ?, 50",
+			FROM `notify`
+			LEFT JOIN `post` ON `post`.`uri-id` = `notify`.`uri-id`
+			WHERE `notify`.`uid` = ?
+			AND `notify`.`msg` != ''
+			AND `notify`.`type` != ?
+			AND $seensql `notify`.`seen`
+			ORDER BY `notify`.`date` $order
+			LIMIT ?, 50",
 			$uid,
-			Notification\Type::INTRO,
```

Comment:
Doesn't this change adds such notifications for all users?

Reply:
Yes, this is the "bad" part of this. Since we fetch the introductions independently, these notifications have been filtered out, but in the case of soapbox/freelove accounts, there's no introduction to match on. I guess I should make it a dynamic array depending on the user's account type.